app-frontend/src/app/components/pages/project/layer/scenes/scenes/index.js

```diff
@@ -167,12 +184,46 @@ class LayerScenesController {
     }
 
     browseScenes() {
-        this.getMap().then(mapWrapper => {
-            const bbox = mapWrapper.map.getBounds().toBBoxString();
-            this.$state.go('project.layer.browse', {bbox});
+        // we cannot just use this.layer here to access its AOI since it might be out of date
+        this.projectService.getProjectLayer(this.projectId, this.layerId).then(resp => {
+            this.layer = resp;
+            if (get(this.layer, 'geometry.type')) {
+                this.getMap().then(mapWrapper => {
+                    const bbox = mapWrapper.map.getBounds().toBBoxString();
+                    this.$state.go('project.layer.browse', { bbox });
+                });
+            } else {
+                this.showRouteToAoiCreateModal();
+            }
         });
     }
 
+    showRouteToAoiCreateModal() {
+        this.modalService
+            .open({
+                component: 'rfFeedbackModal',
+                resolve: {
+                    title: () => 'No AOI defined on this layer',
+                    content: () =>
+                        '<h2>Do you wish to add layer AOI?</h2>' +
+                        '<p>Layer\'s area of interested (AOI) is required to browse scenes. ' +
```

Comment:
One minor phrasing suggestion:

```suggestion
                        '<p>An are of interest (AOI) is required on your layer to browse scenes. ' +
```